tools/msconvert/msconvert_macros.xml

```diff
@@ -523,20 +550,20 @@
 
   <xml name="msconvertOutput">
     <outputs>
-      <data format="mzml" name="output" label="${($input.name[:-4] if $input.name.endswith('.tar') else $input.name).rsplit('.',1)[0]}.${output_type}" >
-        <filter>general_options['multi_run_output']['do_multi_run_output'] == False</filter>
+      <data format="mzml" name="output" label="${tool.name} on ${on_string}" >
```

Comment:
We had an analogous discussion here https://github.com/galaxyproteomics/tools-galaxyp/issues/706#issuecomment-1540075407 and the solution is to use collections. 

We are currently adapting the W4M tools accordingly: https://github.com/workflow4metabolomics/tools-metabolomics/pull/238 https://github.com/workflow4metabolomics/tools-metabolomics/pull/239 .. let me check if the msconvert tool uses `name` somewhere.

> I think the lint is being too aggressive.

We discussed this here: https://github.com/galaxyproject/galaxy/pull/15933#discussion_r1188401691

The problem is that it's hard to check if the filters are mutually exclusive. So we decided to enforce being explicit and forbid outputs of the same name.

Reply:
Who decided this ? This sounds like a bad idea.

Reply:
> I've never liked these `<tool> on <dataset>` names in Galaxy nor that you can only override them by using a workflow

Yes, 💯, such a bad idea to attempt to capture provenance. I'd say though that input.name is also not exactly the right thing to do. I'd just give it a reasonable default, like `${tool.name}: raw data` or whatever is appropriate. If passed a collection the element_identifier will be kept  throughout the analysis.

Reply:
With `${tool.name}: raw data` all msconvert jobs in a history would have the same name ... and I would prefer any of the two other possibilities.

Reply:
I can't really accept that the only way you can keep "sample" names (which in proteomics is usually the filename unless the run is multiplexed like iTRAQ/TMT) is when using collections (which also wouldn't work for multiplexed runs). It really should work just like the command-line. If I put in filename.raw, I should get out filename.mzML. It's not necessarily capturing provenance, but it's behaving like the command-line would and that behavior is often crucial for downstream semantics. There's also WIFF files which contain multiple runs (each of which could be multiplexed, but that's uncommon), so when converting a single WIFF file there needs to be multiple mzML files named according to the run names in the WIFF.

Reply:
I don't disagree, as long as you don't use `{on_string}` which includes the position in the history. The thing about collections though is that it does what you want, keep the identifiers.

> (which also wouldn't work for multiplexed runs)

why is that ? I suppose you'd have a nested collection for things belonging to the same sample ?

Reply:
A nested collection might work as the output of a demultiplexing job. But as long as the samples are multiplexed into one file, representing them as a nested collection would have to point to the same file (possibly with an index indicating which sample is referred to).

Reply:
I just don't see a good reason for the naming behavior of single datasets and collections to be so different. It's not unusual to test a tool or series of tools on just a single file, and I'm not sure why one should need to make a collection of that single file first in order to get sensible naming. TBH I'd even like to see collection member names change depending on their type so it's obvious just from looking at the entry that it's an mzML, or a mzIdentML, or TSV, etc. File extensions are an imperfect but very popular and established solution to the real problem of being able to look at a filename and have an idea of what its contents/purpose is. IMO tool wrappers for command-line tools that use file extensions descriptively should preserve that behavior, whether in single datasets or collections.

Reply:
I think that every automatic naming scheme has its pros and cons and there will always be users who like one more than the other. 

I admit that I disliked Galaxy's default naming scheme at first, but I got used to it .. which does not mean that it's good :)

> It really should work just like the command-line

The problem is that the command line is operated by a human that sets the argument values, for "free text parameters" like filenames the operating human can choose sensible names (i.e. that make sense to her). 

So in some sense, Galaxy works exactly as the command line .. it just uses other names than you would. One thing I like about Galaxy is the consistency of the naming scheme even if it may not be the best in all cases.

> pretty easy to reconstruct the on section in the client

I found it always very convenient to be able to reconstruct this in most cases without clicking. Butm maybe that's the main point: which information do users want to be visible (without the need for additional clicks): 

- which tool generated the data with which input (data sets)
- sample name + file type

Both will "fail" for instance for an analysis of the parameter space of a tool. Then you would like to have the value of a specific parameter in the dataset name, e.g. `BWA with k=23`.

What came to my mind, was how about per user / per history configurable naming schemes?


> ... here's also WIFF files which contain multiple runs ...

That's just another can of worms. Seems that some proteomics datatypes include references to other files. This won't work in Galaxy (out of the box), since Galaxy stores the datasets using completely different names (using some hash as a name).

Reply:
> The problem is that the command line is operated by a human that sets the argument values, for "free text parameters" like filenames the operating human can choose sensible names (i.e. that make sense to her).

In most cases in proteomics, the software always picks a default output name based on the input filename (usually by changing file extension). That's why I think it should be the default naming scheme for Galaxy wrappers that wrap such tools. But of course you're right it has pros and cons.